### ISSUE TITLE
Improve `Rustfmt` integration

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#6145]
Since #5704 there is an option to use `Rustfmt` instead of built-in formatter in the case of **whole file** formatting by:
1. `Code | Reformat Code` (`Ctrl + Alt + L`) action **without text selection**
2. `Code | Reformat File ...` (`Ctrl + Alt + Shift + L`) action with `Whole file` option

- [ ] Enable the option by default #6121
- [ ] Use `Rustfmt` when formatting selected text (we can take a look at https://github.com/rust-lang/rustfmt/issues/3397). Currently, the built-in formatter is still in use.
- [ ] Use `Rustfmt` when formatting is triggered before commit by VCS plugin (if `Reformat Code Before Commit` option is enabled).  Currently, the built-in formatter is still in use.

**Comment by Sherlock-Holo** [comment]
even select the `use rustfmt instead of built-in formatter`, the result is not equal `cargo fmt --`

if can let the formatted result be the same as `cargo fmt --`, it will be excellent

**Comment by vlad20012** [comment]
@Sherlock-Holo Hmm, it's unexpected. Could you provide a minimal example where results differ? Also, maybe your current rustup toolchain has no rustfmt installed?

**Comment by Sherlock-Holo** [comment]
![image](https://user-images.githubusercontent.com/10096425/104404724-508ff880-5596-11eb-85b5-0087fde52c6b.png)
this is the intellij-rust format with `use rustfmt instead of built-in formatter`

after run `cargo fmt --`
![image](https://user-images.githubusercontent.com/10096425/104404835-9351d080-5596-11eb-94e7-2a83b463153a.png)
the codes are changed

**Comment by Sherlock-Holo** [comment]
and when I run `rustfmt -h` it prints
> usage: /home/sherlock/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustfmt [options] <file>...

so I've installed the rustfmt

**Comment by vlad20012** [comment]
@Sherlock-Holo I tried your example, and intellij-rust reformats the code just like `cargo fmt --` in your case. It's weird

**Comment by Sherlock-Holo** [comment]
in clion i use ctrl shift alt l to format codes with whole File option, do you use the same way to format？

**Comment by vlad20012** [comment]
Yep

**Comment by Sherlock-Holo** [comment]
@vlad20012 I've tried rustfmt in `$HOME/.rustup` and provider by `rustup` (I use archlinux and I've installed `rustup` package) both of them format codes as
```rust
fn doit<R>(r: R) -> u64
where
    R: Read,
{
    todo!()
}
```
and even I delete the `/usr/bin/rustfmt` (I try to force clion use the rustfmt which in `$HOME/.rustup`), the clion always format codes as
```rust
fn doit<R>(r: R) -> u64
    where
        R: Read,
{
    todo!()
}
```

**Comment by Sherlock-Holo** [comment]
it seems the `ctrl+alt+shift+l` doesn't use rustfmt even choose `use rustfmt instead of built-in formatter`, because I try `ctrl+alt+l`and it formats codes like rustfmt do
